skimage/filters/thresholding.py

```diff
@@ -236,16 +236,66 @@ def threshold_local(image, block_size, method='gaussian', offset=0,
     return thresh_image - offset
 
 
-def threshold_otsu(image, nbins=256):
+def _validate_image_histogram(image, hist, nbins=None):
+    """Ensure that either image or hist were given, return valid histogram.
+
+    If hist is given, image is ignored.
+
+    Parameters
+    ----------
+    image : array or None
+        Grayscale image.
+    hist : array, 2-tuple of array, or None
+        Histogram, either a 1D counts array, or an array of counts together
+        with an array of bin centers.
+    nbins : int, optional
+        The number of bins with which to compute the histogram, if `hist` is
+        None.
+
+    Returns
+    -------
+    counts : 1D array of float
+        The number of pixels falling in each intensity bin
+    bin_centers : 1D array
+        The value corresponding to the center of each intensity bin.
+
+    Raises
+    ------
+    ValueError : if image and hist are both None
+    """
+    if image is None and hist is None:
+        raise Exception("Either image or hist must be provided.")
+
+    if hist is not None:
+        if isinstance(hist, (tuple, list)):
+            counts, bin_centers = hist
+        else:
+            counts = hist
+            bin_centers = np.arange(counts.size)
+    else:
+        counts, bin_centers = histogram(
+                image.ravel(), nbins, source_range='image'
+            )
+    return counts.astype(float), bin_centers
+
+
+def threshold_otsu(image=None, nbins=256, *, hist=None):
     """Return threshold value based on Otsu's method.
 
+    Either image or hist must be provided. In case hist is given, the actual
```

Comment:
```suggestion
    Either image or hist must be provided. If hist is provided, the actual
```
I find it more direct this way.